More varied OD effects

comma opened the pull request:
-Paracetamoland cold medicine are not doing damage, but still applying 'toxic' effect that prevents self healing.
-Bicard healing arterial bleedings at the cost of causing blood blockage
-Inapro slows you down a bit, and sometimes makes you slur or get drowsy


Could use more ideas
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

ghost commented:
You could fit an exception for Vox on ammonia OD in this one.

comma commented:
oh yeah

ghost commented:
>-Bicard healing arterial bleedings at the cost of causing blood blockage

Bandaged arterial bleedings do literally nothing anyways. This would also make vein-o-fix obsolete.

comma commented:
Fixoveins don't
-damage your internal organs
-deal braindamage since they put you at 80% oxygenation
-need 30+ units of bicard to work

Techhead0 commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm:
```diff
@@ -53,6 +53,10 @@
 		M.adjustToxLoss(removed * 1.5)
 
+/datum/reagent/ammonia/overdose(var/mob/living/carbon/M, var/alien)
+	if(alien != IS_VOX && volume < overdose*6)
```
Meant to be `||` and `>`?

comma replied:
i should stop codig at 4 am

ghost commented:
My bad. I thought it was any amount of Bicaridine, and not an overdose.

PurpleMartinJCK commented on code/__defines/chemistry.dm:
```diff
@@ -44,6 +44,7 @@
 #define CE_BREATHLOSS    "breathloss"   // Breathing depression, makes you need more air
 #define CE_MIND    		 "mindbending"  // Stabilizes or wrecks mind. Used for hallucinations
 #define CE_CRYO 	     "cryogenic"    // Prevents damage from being frozen
+#define CE_BLOCKAGE	     "blockage"     // Gets in the way of blood circulation, higher the worse
```
This effect needs to not work for open circulatory systems (where heart.open == true)

PsiOmegaDelta replied:
Awaiting addressing of this comment.

comma replied:
ugh stupid nabber snowflakes

comma replied:
Actually, how do open circulatory systems work? Isn't it a bunch of tiny vessels? Why would they not be affected by constricting vessels?

PurpleMartinJCK replied:
They don't have a bunch of tiny vessels that can get blocked. Their body cavities have fluids sloshing around their organs and muscles.

comma commented:
don't want to bother with adding GAS features

Techhead0 commented on code/modules/organs/blood.dm:
```diff
@@ -282,6 +282,8 @@ proc/blood_splatter(var/target,var/datum/reagent/blood/source,var/large,var/spra
 			if(PULSE_2FAST, PULSE_THREADY)
 				pulse_mod *= 1.25
 		blood_volume *= max(0.3, (1-(heart.damage / heart.max_damage))) * pulse_mod
+		if(chem_effects[CE_BLOCKAGE])
```
`if(chem_effects[CE_BLOCKAGE] && !heart.open)` should work for Steve's thing. Problem solved.

Techhead0 commented:
Also, with Inapp having relatively benign OD effects, the OD threshold doesn't need to be so high.

comma committed:
More varied OD effects

-Paracetamoland cold medicine are not doing damage, but still applying 'toxic' effect that prevents self healing.
-Bicard healing arterial bleedings at the cost of causing blood blockage
-Inapro slows you down a bit, and sometimes makes you slur or get drowsy

comma committed:
Makes vox not OD on ammonia as easily
